inception-external-search-solr/src/test/java/SolrSearchProviderTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020
+ * ENP-China, Aix-Marseille University
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import de.tudarmstadt.ukp.inception.externalsearch.ExternalSearchResult;
+import de.tudarmstadt.ukp.inception.externalsearch.model.DocumentRepository;
+import de.tudarmstadt.ukp.inception.externalsearch.solr.SolrSearchProvider;
+import de.tudarmstadt.ukp.inception.externalsearch.solr.SolrSearchProviderFactory;
+import org.junit.Before;
+import org.junit.Test;
+import de.tudarmstadt.ukp.inception.externalsearch.solr.traits.SolrSearchProviderTraits;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+public class SolrSearchProviderTest {
+    private SolrSearchProvider sut;
+    private DocumentRepository repo;
+    private SolrSearchProviderTraits traits;
+
+    @Before
+    public void setup()
+    {
+        sut = new SolrSearchProvider();
+
+        repo = new DocumentRepository("test", null);
+
+        traits = new SolrSearchProviderTraits();
+        traits.setRemoteUrl("http://localhost:8983");
```

Comment:
Hm, the test does not seem to set up a SOLR instance to which it can talk - I wonder if this works.

Reply:
You are correct, the user has to be running a local instance of solr. I believe the test in the elasticsearch module also rely on external index, and I'm not sure this test deserves that we add the dependency to the solr server.

I asked @alexis-houpert to configure the test so that it relies on the same collections as in the solr tutorial, it should be easy enough to run and to document in a readme file

Reply:
The test in the ElasticSearch module is not run - it is marked with an `@Ignore` because the server referenced there is not generally available. I wouldn't mind adding an `@Ignore` to the Solr test as well if spinning up an embedded test instance is too much effort.

Reply:
Ok, we did that. and we set up the test as an example which rely on the collection from Solr tutorial. Documentation will be updated soon. 
(Other things should be ok too in the last update, don't hesitate to assign change requests to alexis directly)